x-pack/legacy/plugins/ml/server/routes/system.ts

```diff
@@ -0,0 +1,243 @@
+  router,
+  xpackMainPlugin,
+  spacesPlugin,
+  cloud,
+}: RouteInitialization) {
+  async function getNodeCount(context: RequestHandlerContext) {
+    const filterPath = 'nodes.*.attributes';
+    const resp = await context.ml!.mlClient.callAsInternalUser('nodes.info', {
+      filterPath,
+    });
+
+    let count = 0;
+    if (typeof resp.nodes === 'object') {
+      Object.keys(resp.nodes).forEach(k => {
+        if (resp.nodes[k].attributes !== undefined) {
+          const maxOpenJobs = resp.nodes[k].attributes['ml.max_open_jobs'];
+          if (maxOpenJobs !== null && maxOpenJobs > 0) {
+            count++;
+          }
+        }
+      });
+    }
+    return { count };
+  }
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @api {post} /api/ml/_has_privileges Check privileges
+   * @apiName HasPrivileges
+   * @apiDescription Checks if the user has required privileges
+   */
+  router.post(
+    {
+      path: '/api/ml/_has_privileges',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        let upgradeInProgress = false;
+        try {
+          const info = await context.ml!.mlClient.callAsCurrentUser('ml.info');
+          // if ml indices are currently being migrated, upgrade_mode will be set to true
+          // pass this back with the privileges to allow for the disabling of UI controls.
+          upgradeInProgress = info.upgrade_mode === true;
+        } catch (error) {
+          // if the ml.info check fails, it could be due to the user having insufficient privileges
+          // most likely they do not have the ml_user role and therefore will be blocked from using
+          // ML at all. However, we need to catch this error so the privilege check doesn't fail.
+          if (error.status === 403) {
+            mlLog.info(
+              'Unable to determine whether upgrade is being performed due to insufficient user privileges'
+            );
+          } else {
+            mlLog.warn('Unable to determine whether upgrade is being performed');
+          }
+        }
+
+        if (isSecurityDisabled(xpackMainPlugin)) {
+          // if xpack.security.enabled has been explicitly set to false
+          // return that security is disabled and don't call the privilegeCheck endpoint
+          return response.ok({
+            body: {
+              securityDisabled: true,
+              upgradeInProgress,
+            },
+          });
+        } else {
+          const body = request.body;
+          const resp = await context.ml!.mlClient.callAsCurrentUser('ml.privilegeCheck', { body });
+          resp.upgradeInProgress = upgradeInProgress;
+          return response.ok({
+            body: resp,
+          });
+        }
+      } catch (error) {
+        return response.customError(wrapError(error));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @api {post} /api/ml/ml_capabilities Check ML capabilities
```

Comment:
This should be `get`

Reply:
https://github.com/elastic/kibana/pull/57835/commits/608f48dbacc75c6842b84eac804650de03190c1b